packages/payment-experience-api/src/api/getUpdateCardFormParametersController.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  AbstractController,
+  Data,
+  ExperienceError,
+  logger,
+  StatusCode,
+  ValidatedRequest,
+} from '@verkkokauppa/core'
+import type { Response } from 'express'
+import * as yup from 'yup'
+import {
+  getUpdatePaytrailCardFormParams,
+  PaymentGateway,
+} from '@verkkokauppa/payment-backend'
+import { getOrder } from '@verkkokauppa/order-backend'
+import { parseMerchantIdFromFirstOrderItem } from '@verkkokauppa/configuration-backend'
+
+const requestSchema = yup.object().shape({
+  params: yup.object().shape({
+    orderId: yup.string().required(),
+  }),
+  headers: yup.object().shape({
+    user: yup.string().required(),
+  }),
+})
+
+export class GetUpdateCardFormParametersController extends AbstractController<
+  typeof requestSchema
+> {
+  protected readonly requestSchema = requestSchema
+
+  protected async implementation(
+    req: ValidatedRequest<typeof requestSchema>,
+    res: Response
+  ): Promise<any> {
+    const {
+      params: { orderId },
+      headers: { user },
+    } = req
+
+    logger.debug(`Fetch updateCardFormParameters ${orderId}`)
+
+    const order = await getOrder({ orderId, user })
```

Comment:
Is there a reason why getCardFormParameters uses getOrderAdmin and this one uses getOrder?

Reply:
Good catch, could you change getCardFormParameters to use normla getOrder function too with user id?